### PR TITLE
Writing prompts: adds basic UI to blogging prompts block

### PR DESCRIPTION
**Pull request by creativecoder**
## Proposed changes:

Fills out the blogging prompts block with attributes, edit/save functions, styles, and initial controls.

Follow up to https://github.com/Automattic/jetpack/pull/29013

Requires https://github.com/Automattic/jetpack/pull/29182

image

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pe7F0s-uo-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Follow the testing instructions in https://github.com/Automattic/jetpack/pull/29182 to set up the API endpoint 
- Enable beta blocks (`define( 'JETPACK_BETA_BLOCKS', true );`)
- Open the editor and insert a writing prompt block.
- Save the post and view the front-end of the block.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (roo2) [inline projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js]: I think this will throw an error for high month values, calling the api with `--012-01` returns a 400 error.

Oh or does `slice( -2 )` handle that?

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/save.js
Comment (roo2) [inline projects/plugins/jetpack/extensions/blocks/blogging-prompt/save.js]: prompt_id comes through as undefined here which leads to the generated HTML having a broken link

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (roo2) [inline projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js]: where does promptId get set? it comes through as todays prompt id, but not from the `answer_prompt` query string as I'd expect

**Review by roo2 — COMMENTED** [review]
Just a few issues with prompt_id, I think it should default to what was passed in `answer_prompt` at first, and then, if a user loads an existing post, it should be whatever was saved for that block

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (creativecoder) [reply projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js]: `.slice` does handle that, but I'll switch to using `padStart`, which hopefully makes it a little more clear.

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (creativecoder) [reply projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js]: For a block that's newly inserted, the promptId will be empty be default, and then filled in with the most recent prompt for today. For an existing block, the promptId will be stored, and fresh data pulled in case information about prompt answers has changed.

I'm not handling the integration with the existing `blogging-prompts` extension yet, which uses the answer_prompt param  (not to be confused with this `blogging-prompt` beta extension being used to develop this new block).

To keep the PRs more manageable, I'm developing the block on it's own. When we're happy with it, I'll merge the two extensions, use the answer_prompt param to insert a writing prompt block with the promptId pre-filled, and mark as experimental, to release to WP.com sites.

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/save.js
Comment (creativecoder) [reply projects/plugins/jetpack/extensions/blocks/blogging-prompt/save.js]: That's a bug, thanks for catching it! Now fixed.

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (roo2) [reply projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js]: ok, that makes sense 👍

**Review by roo2 — DISMISSED** [review]
Retested and LGTM

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js:107]: Maybe in a follow-up PR, we could make this an `ExternalLink`?

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/save.js
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/blogging-prompt/save.js:34]: Same as above, I think this could be an external link.

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/style.scss
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/blogging-prompt/style.scss:2]: I think we could use variables in this file, by importing `@import '@automattic/jetpack-base-styles/gutenberg-base-styles';`

See https://github.com/Automattic/jetpack/blob/4175424728e0e168f4155084782d750c4683497f/projects/plugins/jetpack/extensions/README.md#L345-L352 for more info.

**Review by jeherve — DISMISSED** [review]
This looks good to me. I left a couple of notes below, but nothing that should be a blocker.

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js
Comment (creativecoder) [reply projects/plugins/jetpack/extensions/blocks/blogging-prompt/edit.js:107]: Happy to do that in a follow-up, thanks for the suggestion!

File: projects/plugins/jetpack/extensions/blocks/blogging-prompt/style.scss
Comment (creativecoder) [reply projects/plugins/jetpack/extensions/blocks/blogging-prompt/style.scss:2]: Nice idea, I will check in with @ollierozdarz who worked on initial designs for the block and handle that in a follow-up.

**Comment by creativecoder** [comment]
Opps, need another approval after resolving the merge conflict.

**Review by jeherve — APPROVED** [review]

**Pull request merged**
